build_tools/provision.py

```diff
@@ -0,0 +1,206 @@
+    _untar_files(output_dir, destination)
+
+
+def retrieve_artifacts_by_ci(args):
+    """
+    If the user requested TheRock artifacts by CI (runner ID), this function will retrieve those assets
+    """
+    runner_id = args.runner_id
+    output_dir = args.output_dir
+    amdgpu_family = args.amdgpu_family
+    log(f"Retrieving artifacts for runner ID {runner_id}")
+    if not s3_bucket_exists(runner_id):
+        log(f"S3 artifacts for {runner_id} does not exist. Exiting...")
+        return
+
+    args.all = True
+
+    # Retrieving base and all math-lib tar artifacts and downloading them to output_dir
+    retrieve_base_artifacts(args, runner_id, output_dir)
+    retrieve_enabled_artifacts(args, True, amdgpu_family, runner_id, output_dir)
+
+    # Flattening artifacts from .tar* files then removing .tar* files
+    log(f"Untar-ing artifacts for {runner_id}")
+    output_dir_path = Path(output_dir).resolve()
+    tar_file_paths = list(output_dir_path.glob("*.tar.*"))
+    flattener = ArtifactPopulator(
+        output_path=output_dir_path, verbose=True, flatten=True
+    )
+    flattener(*tar_file_paths)
+    for file_path in tar_file_paths:
+        file_path.unlink()
+
+    log(f"Retrieved artifacts for runner ID {runner_id}")
+
+
+def retrieve_artifacts_by_release(args):
+    """
+    If the user requested TheRock artifacts by release tag (release ID), this function will retrieve those assets
+    """
+    release_id = args.release_id
+    output_dir = args.output_dir
+    amdgpu_family = args.amdgpu_family
+    log(f"Retrieving artifacts for release ID {release_id}")
+    asset_data = _get_github_release_assets(release_id, amdgpu_family)
+    if not asset_data:
+        log(f"GitHub release asset for {release_id} not found. Exiting...")
+        return
+    _download_github_release_asset(asset_data, output_dir)
+    log(f"Retrieving artifacts for runner ID {release_id}")
+
+
+def run(args):
+    log("### Provisioning TheRock 🪨 ###")
+    _create_output_directory(args)
+    if args.runner_id:
+        retrieve_artifacts_by_ci(args)
+
+    if args.release_id:
+        retrieve_artifacts_by_release(args)
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(prog="provision")
+    parser.add_argument(
+        "--output-dir",
+        type=str,
+        default="./therock-build",
+        help="Path of the output directory for TheRock",
+    )
+
+    parser.add_argument(
+        "--amdgpu-family",
+        type=str,
+        default="gfx94X-dcgpu",
+        help="AMD GPU family to provision (please refer to this: https://github.com/ROCm/TheRock/blob/59c324a759e8ccdfe5a56e0ebe72a13ffbc04c1f/cmake/therock_amdgpu_targets.cmake#L44-L81 for family choices)",
+    )
+
+    # This mutually exclusive group will ensure that only one argument is present
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "--runner-id", type=str, help="GitHub runner ID of TheRock to provision"
```

Comment:
```suggestion
        "--run-id", type=str, help="GitHub run ID of TheRock to provision"
```
I would suggest to rename to `run-id` as this is not the runner but a specific run, not necessarily associated with a specific runner.